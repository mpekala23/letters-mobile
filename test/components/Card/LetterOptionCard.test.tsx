import React from 'react';
import { fireEvent, render, toJSON } from '@testing-library/react-native';
import LetterOptionCard, {
  LetterTypes,
} from '../../../src/components/Card/LetterOptionCard.react';

const setup = (propOverrides = {}) => {
  const props = {
    type: LetterTypes.PostCards,
    onPress: jest.fn(),
    ...propOverrides,
  };
  return {
    ...render(<LetterOptionCard {...props} />),
    props,
  };
};

describe('Letter Option Card component', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('should fire onPress() on a press', () => {
    const { props, getByText } = setup();
    fireEvent.press(getByText('Post cards'));
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });

  it('should display title', () => {
    const { getByText } = setup();
    expect(getByText('Post cards')).toBeDefined();
  });

  it('should display description', () => {
    const { getByText } = setup();
    expect(getByText('1 photo, 300 characters')).toBeDefined();
  });
});
