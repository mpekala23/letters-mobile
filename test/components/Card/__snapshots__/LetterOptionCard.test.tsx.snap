// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letter Option Card component should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 6,
          "marginVertical": 6,
          "padding": 15,
          "width": "100%",
        },
        Object {
          "flexDirection": "column",
          "height": 110,
          "justifyContent": "space-between",
        },
        Object {
          "elevation": 4,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2.62,
        },
        undefined,
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={68}
      bbWidth={111}
      fill="none"
      focusable={false}
      height={68}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "bottom": 0,
            "position": "absolute",
            "right": 0,
          },
          Object {
            "flex": 0,
            "height": 68,
            "width": 111,
          },
        ]
      }
      vbHeight={68}
      vbWidth={111}
      width={111}
      xml="<svg width=\\"111\\" height=\\"68\\" viewBox=\\"0 0 111 68\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <path
    d=\\"M54.9922 13.5328L78.4832 18.8756L54.9922 13.5328ZM52.855 22.9292L62.2514 25.0663L52.855 22.9292ZM60.3263 44.3934L53.067 32.8599L46.0197 31.257C44.7737 30.9736 43.6913 30.2068 43.0106 29.1254C42.3299 28.0439 42.1067 26.7363 42.3901 25.4903L46.6643 6.69746C46.9477 5.45143 47.7145 4.369 48.796 3.68831C49.8775 3.00762 51.1851 2.78443 52.4311 3.06783L85.3185 10.5478C86.5645 10.8312 87.6469 11.598 88.3276 12.6795C89.0083 13.7609 89.2315 15.0685 88.9481 16.3146L84.6739 35.1073C84.3905 36.3534 83.6237 37.4358 82.5422 38.1165C81.4607 38.7972 80.1531 39.0204 78.9071 38.737L71.8598 37.1341L60.3263 44.3934Z\\"
    stroke=\\"#448AF3\\"
    strokeWidth=\\"3\\"
    strokeLinecap=\\"round\\"
    strokeLinejoin=\\"round\\"
  />
  <path
    d=\\"M10.5821 49.9328L23.9331 42.1375L10.5821 49.9328ZM13.7002 55.2732L19.0406 52.1551L13.7002 55.2732ZM26.6119 62.0565L18.1534 59.8341L14.148 62.1727C13.4399 62.5862 12.5964 62.7014 11.8033 62.493C11.0101 62.2846 10.3323 61.7697 9.91878 61.0615L3.68258 50.3807C3.2691 49.6725 3.15387 48.829 3.36226 48.0359C3.57064 47.2428 4.08556 46.5649 4.79375 46.1514L23.4853 35.2381C24.1934 34.8246 25.0369 34.7094 25.83 34.9177C26.6232 35.1261 27.301 35.641 27.7145 36.3492L33.9507 47.0301C34.3642 47.7383 34.4794 48.5817 34.271 49.3748C34.0627 50.168 33.5477 50.8459 32.8395 51.2594L28.8342 53.5979L26.6119 62.0565Z\\"
    stroke=\\"#448AF3\\"
    strokeWidth=\\"3\\"
    strokeLinecap=\\"round\\"
    strokeLinejoin=\\"round\\"
  />
  <path
    d=\\"M92.2365 57.7015L105.219 57.5161L92.2365 57.7015ZM92.3107 62.8947L97.5038 62.8205L92.3107 62.8947ZM98.9505 73.1882L93.6832 68.0692L89.7883 68.1249C89.0997 68.1347 88.4353 67.8706 87.9414 67.3906C87.4475 66.9106 87.1645 66.254 87.1547 65.5654L87.0063 55.1791C86.9965 54.4905 87.2606 53.8261 87.7406 53.3322C88.2206 52.8383 88.8771 52.5553 89.5658 52.5455L107.742 52.2859C108.43 52.2761 109.095 52.5402 109.589 53.0202C110.083 53.5002 110.366 54.1567 110.375 54.8454L110.524 65.2316C110.534 65.9203 110.269 66.5846 109.789 67.0785C109.309 67.5724 108.653 67.8554 107.964 67.8653L104.069 67.9209L98.9505 73.1882Z\\"
    stroke=\\"#448AF3\\"
    strokeWidth=\\"3\\"
    strokeLinecap=\\"round\\"
    strokeLinejoin=\\"round\\"
  />
</svg>"
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M54.9922 13.5328L78.4832 18.8756L54.9922 13.5328ZM52.855 22.9292L62.2514 25.0663L52.855 22.9292ZM60.3263 44.3934L53.067 32.8599L46.0197 31.257C44.7737 30.9736 43.6913 30.2068 43.0106 29.1254C42.3299 28.0439 42.1067 26.7363 42.3901 25.4903L46.6643 6.69746C46.9477 5.45143 47.7145 4.369 48.796 3.68831C49.8775 3.00762 51.1851 2.78443 52.4311 3.06783L85.3185 10.5478C86.5645 10.8312 87.6469 11.598 88.3276 12.6795C89.0083 13.7609 89.2315 15.0685 88.9481 16.3146L84.6739 35.1073C84.3905 36.3534 83.6237 37.4358 82.5422 38.1165C81.4607 38.7972 80.1531 39.0204 78.9071 38.737L71.8598 37.1341L60.3263 44.3934Z"
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4282682099}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
        />
        <RNSVGPath
          d="M10.5821 49.9328L23.9331 42.1375L10.5821 49.9328ZM13.7002 55.2732L19.0406 52.1551L13.7002 55.2732ZM26.6119 62.0565L18.1534 59.8341L14.148 62.1727C13.4399 62.5862 12.5964 62.7014 11.8033 62.493C11.0101 62.2846 10.3323 61.7697 9.91878 61.0615L3.68258 50.3807C3.2691 49.6725 3.15387 48.829 3.36226 48.0359C3.57064 47.2428 4.08556 46.5649 4.79375 46.1514L23.4853 35.2381C24.1934 34.8246 25.0369 34.7094 25.83 34.9177C26.6232 35.1261 27.301 35.641 27.7145 36.3492L33.9507 47.0301C34.3642 47.7383 34.4794 48.5817 34.271 49.3748C34.0627 50.168 33.5477 50.8459 32.8395 51.2594L28.8342 53.5979L26.6119 62.0565Z"
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4282682099}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
        />
        <RNSVGPath
          d="M92.2365 57.7015L105.219 57.5161L92.2365 57.7015ZM92.3107 62.8947L97.5038 62.8205L92.3107 62.8947ZM98.9505 73.1882L93.6832 68.0692L89.7883 68.1249C89.0997 68.1347 88.4353 67.8706 87.9414 67.3906C87.4475 66.9106 87.1645 66.254 87.1547 65.5654L87.0063 55.1791C86.9965 54.4905 87.2606 53.8261 87.7406 53.3322C88.2206 52.8383 88.8771 52.5553 89.5658 52.5455L107.742 52.2859C108.43 52.2761 109.095 52.5402 109.589 53.0202C110.083 53.5002 110.366 54.1567 110.375 54.8454L110.524 65.2316C110.534 65.9203 110.269 66.5846 109.789 67.0785C109.309 67.5724 108.653 67.8554 107.964 67.8653L104.069 67.9209L98.9505 73.1882Z"
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4282682099}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins-Bold",
          },
          Object {
            "color": "#282828",
            "fontSize": 24,
          },
        ]
      }
    >
      Photos
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#6E757C",
            "fontSize": 18,
          },
          Object {
            "maxWidth": "65%",
          },
        ]
      }
    >
      Glossy photo & short caption
    </Text>
  </TouchableOpacity>
</View>
`;
